### ISSUE TITLE
Plugin resolved from custom plugin repository with transitive dependencies in other repositories fails

### ISSUE THREAD

**Issue by sdeleuze** [gradle/gradle#3210]
I need to use Gradle custom `pluginManagement` as documented in [Gradle Kotlin DSL documentation](https://github.com/gradle/kotlin-dsl/blob/2fc2bc1b10980d995809c171b0f02158c4336a80/doc/getting-started/Configuring-Plugins.md) because some plugin I am using are not yet on Gradle plugin portal (Junit 5 and Spring Boot `2.0.0.M5`).

It works when all dependencies are in Gradle cache but fails with an empty cache. It can be reproduced with [this branch](https://github.com/sdeleuze/spring-kotlin-fullstack/tree/gradle-cache-issue) of my repository (and more specifically my [settings.gradle](https://github.com/sdeleuze/spring-kotlin-fullstack/blob/gradle-cache-issue/settings.gradle) file):

```
rm -rf ~/.gradle/caches
git clone https://github.com/sdeleuze/spring-kotlin-fullstack.git`
git checkout gradle-cache-issue
./gradlew build
```

Both Gradle `4.2.1` and `4.3.rc-1` have this issue, probably previous versions as well. This issue is currently preventing me to use Gradle for my upcoming [Devoxx BE livecoding](https://cfp.devoxx.be/2017/talk/RYZ-3811/From_Spring_Boot_1_in_Java_to_Spring_Boot_2_in_Kotlin).

**Comment by eskatos** [comment]
This fails with:

```
Resource missing. [HTTP GET: https://repo.spring.io/milestone/com/fasterxml/jackson/jackson-bom/2.9.1/jackson-bom-2.9.1.pom]
Resource missing. [HTTP HEAD: https://repo.spring.io/milestone/com/fasterxml/jackson/jackson-bom/2.9.1/jackson-bom-2.9.1.jar]


FAILURE: Build failed with an exception.

* Where:
Build file '/Users/paul/src/local/gradle/spring-kotlin-fullstack/backend/build.gradle.kts' line: 2

* What went wrong:
Plugin [id: 'org.springframework.boot', version: '2.0.0.M5', artifact: 'org.springframework.boot:spring-boot-gradle-plugin:2.0.0.M5'] was not found in any of the following sources:

- Gradle Core Plugins (plugin is not in 'org.gradle' namespace)
- Gradle Central Plugin Repository (explicit artifact coordinates are not supported by this source)
- maven(https://jcenter.bintray.com/) (Could not resolve plugin artifact 'org.springframework.boot:spring-boot-gradle-plugin:2.0.0.M5')
- maven(https://repo.spring.io/milestone) (Could not resolve plugin artifact 'org.springframework.boot:spring-boot-gradle-plugin:2.0.0.M5')

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
```

The resolved plugin has transitive dependencies that can be found in a different repository than the one where the plugin artifact was found. But it seems plugin resolution stick with the repository where the plugin artifact was found and can't resolve the transitive dependencies.

**Comment by oehme** [comment]
A simple workaround would be to look up the marker non-transitively [here](https://github.com/gradle/gradle/blob/master/subprojects/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/ArtifactRepositoryPluginResolver.java#L99).

**Comment by eskatos** [comment]
This has been delivered as part of #3347 and will be included in Gradle 4.4-RC1
See https://github.com/gradle/gradle/commit/46675fb8df66325ce42b6f3af6e62bc5d25ef167#diff-c601f5d1d522fb3c362fd8313e0ddbd7R301

Closing

**Comment by sdeleuze** [comment]
Thanks!
